File: 2022/d1/d1a.py
import sys

answer=0
total=0

for line in sys.stdin:
    line=line.rstrip()
    if line == "":
       answer=max(total, answer)
       total=0
    else:
        total+=int(line)

print("The highest calories by any elf is {ans}".format(ans=answer))

File: 2022/d1/d1a.pl
#!/usr/bin/perl

my $answer=0;
my $total=0;

while( <STDIN> ) {
    chomp;

    if($_ eq "") {
        $answer=$total if($total>$answer);
        $total=0;
    } else {
        $total+=$_;
    }
}

printf("The highest calories by any elf is %d\n", $answer);